test/integration/registerable_test.rb

```diff
@@ -1,6 +1,8 @@
 require 'test_helper'
 
 class RegistrationTest < Devise::IntegrationTest
+  uses_transaction :test_a_guest_user_should_receive_the_confirmation_instructions_from_the_default_mailer
```

Comment:
@allenwq You're right. I agree. Maybe the option of using the gem is better.

I think we can at least, can not use the gem configuration if we are on Rails 5 and see what happens.

You can chan the Rails 5 using a code like this: https://github.com/plataformatec/devise/blob/69bee06ceee6280b54304928bb6e55c5064abad8/test/rails_app/app/controllers/home_controller.rb#L23

Then, we can let a "TODO" to remove the gem and code when we stop to support versions below than Rails 5.